### PR TITLE
Use LineColumn type.

### PR DESCRIPTION
**Pull request by chriseth**
This is a refactor that documents the LineColumn type and uses it in one additional function.

### CODE REVIEW

File: liblangutil/SourceLocation.h [hunk 1]
```diff
@@ -119,4 +119,24 @@ SourceLocation parseSourceLocation(
...
+ */
+struct LineColumn
+{
+	/// Line value, can be between zero and number of `\n` characters in the source file.
+	int line = {-1};
+	/// Column value, can be between zero and number of characters in the line (inclusive).
+	int column = {-1};
+
+	LineColumn() = default;
+	explicit LineColumn(std::tuple<int, int> const& _t): line{std::get<0>(_t)}, column{std::get<1>(_t)} {}
```
Comment (chriseth) [inline liblangutil/SourceLocation.h]: I think this is actually not needed anymore.

File: liblangutil/SourceLocation.h [hunk 2]
```diff
@@ -119,4 +119,23 @@ SourceLocation parseSourceLocation(
...
+ */
+struct LineColumn
+{
+	/// Line value, can be between zero and number of `\n` characters in the source file.
+	int line = {-1};
+	/// Column value, can be between zero and number of characters in the line (inclusive).
+	int column = {-1};
+
+	LineColumn() = default;
+	explicit LineColumn(int _line, int _column): line(_line), column(_column) {}
```
Comment (cameel) [inline liblangutil/SourceLocation.h:137]: Does it even need a constructor now that arguments match struct fields and are no longer a tuple?

File: liblangutil/SourceLocation.h [hunk 3]
```diff
@@ -119,4 +119,23 @@ SourceLocation parseSourceLocation(
...
+ * Both line and column are zero-based.
+ * If used as a range, the second location is considered exclusive.
+ * Negative values are invalid.
+ */
+struct LineColumn
+{
+	/// Line value, can be between zero and number of `\n` characters in the source file.
+	int line = {-1};
+	/// Column value, can be between zero and number of characters in the line (inclusive).
+	int column = {-1};
```
Comment (cameel) [inline liblangutil/SourceLocation.h]: It's just an `int`. Why use an initializer list?
```suggestion
	/// Line value, can be between zero and number of `\n` characters in the source file.
	int line = -1;
	/// Column value, can be between zero and number of characters in the line (inclusive).
	int column = -1;
```

**Review by cameel — DISMISSED** [review]
I have two minor remarks but looks good anyway.

File: liblangutil/SourceLocation.h (hunk 1 above)
Comment (cameel) [reply liblangutil/SourceLocation.h]: Yeah, I think it's not needed.

**Comment by cameel** [comment]
By the way, nice PR number :)

**Comment by chriseth** [comment]
Haha, I was actually just wondering today if this number has already turned up :)

File: liblangutil/SourceLocation.h (hunk 2 above)
Comment (chriseth) [reply liblangutil/SourceLocation.h:137]: Yes, but can we have default values and prevent construction from just a single value?

Comment (cameel) [reply liblangutil/SourceLocation.h:137]: True, that's a good argument for keeping the constructor.

**Review by cameel — APPROVED** [review]

**Pull request merged**